internal/gatewayapi/route.go

```diff
@@ -26,6 +26,11 @@ type RoutesTranslator interface {
 	ProcessUDPRoutes(udpRoutes []*v1alpha2.UDPRoute, gateways []*GatewayContext, resources *Resources, xdsIR XdsIRMap) []*UDPRouteContext
 }
 
+const (
+	GRPCServiceRegexPattern = "(?i)\\.?[a-z_][a-z_0-9]*(\\.[a-z_][a-z_0-9]*)*"
```

Comment:
why are there two different types here, isnt the only use case needed is when we want to match on method but service is empty ?

Reply:
we need `GRPMethodRegexPattern` to support a use case when `service` is specified without `method` and `type: RegularExpression`. envoy's prefix is used when `type` is empty or `type: Exact`